### PR TITLE
Edit STV algorithm to eliminate zero vote candidates only once a non-zero vote elimination occurs

### PR DESCRIPTION
**Pull request by fairvotereform**
Hi Trey,

I've looked at the STV algorithm and made some changes that should make it behave as we want (only eliminating zero vote candidates once an elimination round is reached). I've got the function working and tested on my end. Two things though: 

- Would you be able to change the call to `calculateResults` to include the "choices" variable? I changed the stv function call to accept the "choices" variable as a third argument. I'm not familiar with firebase and not exactly sure how to access that variable.

- Part of what's returned by the stv function is a field called "eleminated" that contains dictionaries for each eliminated candidate. Each dictionary include the eliminated candidate's name, the round eliminated, and a "from" field that indicates which candidate was eliminated in that round, if there were multiple possible elimination candidates with the same "losing" round total.  I wasn't exactly sure how to fill this in for the zero-vote candidates when they are eliminated because I'm not sure how you use this field later on. Right now I fill it with a -1 for zero-vote eliminations. Let me know if my change here messes things up.

Also, let me know if I should pull request into a different branch.

Best,
Chris

### CODE REVIEW

**Comment by fairvotereform** [comment]
Hi Trey,

I saw your note about the display order in your last email. I added the info necessary for the display order to the stv function output. The "elected" output is no longer just a list and now contains a dictionary with "name", "round", and "votes". The "eleminated output variable contains the same info now too. (The votes can be used as tie-breakers for candidates elected or eliminated in the same round). Let me know if this works.

Best,
Chris

**Comment by ggordn3r** [comment]
Thanks, Chris. Just to keep everybody on the same page, this is what I said in the email you're referencing:

_Unfortunately, ordering by final round isn't going to do what you want. In the final round of the algorithm, only 2-3 choices have any votes at all, making the sorting of all eliminated candidates a total guess. It sounds like what you really want is a perfect sorting in place order, but the algorithm doesn't provide this. All we have to work with is the number of votes for candidates in the running in any given round. Thus, the two options within scope are A) Sort descending by Round 1 (the current implementation) or B) move winners to top and eliminated to bottom in each round (the implementation during beta). I've asked Stephen and Ralph to enable option B this week for comparison._

So basically it would be a double-sort:
- winners go to the top of the chart in order of election (i.e. ascending by round)
- eliminated go to the bottom of the chart in order of elimination (i.e. descending by round)

I think we're on a time constraint to implement that before we need to push, but I will at least add it to the backlog.
